File: bamboo-sample/src/main/java/com/yuluyao/logger/MainActivity.java
package com.yuluyao.logger;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import yuluyao.logger.Logger;

public class MainActivity extends AppCompatActivity {

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    TextView tvHello = (TextView) findViewById(R.id.tvHello);
    tvHello.setOnClickListener(new View.OnClickListener() {
      @Override public void onClick(View v) {
        Logger.i("点击了这里");
      }
    });
    Logger.v("hello world!");
    Logger.d("hello world!");
    Logger.i("hello world!");
    Logger.w("hello world!");
    Logger.e("hello world!");
  }
}

File: sample/src/main/java/com/yuluyao/logger/MainActivity.kt
package com.yuluyao.logger

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import yuluyao.logger.Logger

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    tvHello?.setOnClickListener {
      Logger.v("my message")
      Logger.v("tag", "my message")

      val tag: String? = "tag-string-no-null"
      Logger.v(tag, "my message")

      val tag2: String? = null
      Logger.v(tag2, "my message")

      Logger.v(this@MainActivity, "my message")
      Logger.v(this, "my message")
    }
  }
}